tools/export.py

```diff
@@ -199,36 +161,60 @@ def stub_roi_feature_extractor(model, extractor_name):
                     extractor[i] = ROIFeatureExtractorStub(extractor[i])
 
 
+def get_fake_input(cfg, orig_img_shape=(128, 128, 3), device='cuda'):
+    test_pipeline = [LoadImage()] + cfg.data.test.pipeline[1:]
+    test_pipeline = Compose(test_pipeline)
+    data = dict(img=np.zeros(orig_img_shape, dtype=np.uint8))
+    data = test_pipeline(data)
+    data = scatter(collate([data], samples_per_gpu=1), [device])[0]
+    return data
+
+
 def main(args):
-    model = init_detector(args.config, args.checkpoint, device='cpu')
-    model.eval().cuda()
     torch.set_default_tensor_type(torch.FloatTensor)
+    model = init_detector(args.config, args.checkpoint, device='cpu')
+    model.eval()
+    model.cuda()
+    device = next(model.parameters()).device
+    cfg = model.cfg
+    fake_data = get_fake_input(cfg, device=device)
 
-    if not args.no_stubs:
+    if args.target == 'openvino' and not args.alt_ssd_export:
         stub_anchor_generator(model, 'rpn_head')
         stub_anchor_generator(model, 'bbox_head')
         stub_roi_feature_extractor(model, 'bbox_roi_extractor')
         stub_roi_feature_extractor(model, 'mask_roi_extractor')
+
+    onnx_model_path = osp.join(args.output_dir,
+                               osp.splitext(osp.basename(args.config))[0] + '.onnx')
     
-    image = np.zeros((128, 128, 3), dtype=np.uint8)
     with torch.no_grad():
-        export(model, image, export_name=args.model, opset=10)
-
-    # add_node_names(args.model)
+        export_to_onnx(model, fake_data, export_name=onnx_model_path, opset=10, alt_ssd_export=args.alt_ssd_export)
+        add_node_names(onnx_model_path)
+        print(f'ONNX model has been saved to "{onnx_model_path}"')
 
-    if args.check:
-        check_onnx_model(args.model)
+    if args.target == 'openvino':
+        input_shape = list(fake_data['img'][0].shape)
+        if args.input_shape:
+            input_shape = [1, 3, *args.input_shape]
+        export_to_openvino(cfg, onnx_model_path, args.output_dir, input_shape)
+    else:
+        check_onnx_model(onnx_model_path)
 
 
 def parse_args():
-    parser = argparse.ArgumentParser(description='Export model to ONNX')
+    parser = argparse.ArgumentParser(description='Export model')
     parser.add_argument('config', help='test config file path')
     parser.add_argument('checkpoint', help="path to file with model's weights")
-    parser.add_argument('model', help='path to output onnx model file')
-    parser.add_argument('--check', action='store_true',
-                        help='check that resulting onnx model is valid')
-    parser.add_argument('--no_stubs', action='store_true',
-                        help='disable all exporting stubs')
+    parser.add_argument('output_dir', help='path to directory to save exported models in')
+
+    subparsers = parser.add_subparsers(title='target', dest='target', help='target model format')
```

Comment:
please add required=True, otherwise non-informative exception is thrown

Reply:
Fixed.